### PR TITLE
Review customers package. close #761

### LINKED ISSUE #761: Review customer package

### PR DESCRIPTION
**Pull request by schleichardt**
@lauraluiz please review

### CODE REVIEW

File: sphere-models/src/it/java/io/sphere/sdk/customers/commands/CustomerPasswordResetCommandTest.java [hunk 1]
```diff
@@ -1,26 +1,46 @@
...
             final String newPassword = "newPassword";
-            final Customer updatedCustomer = execute(CustomerPasswordResetCommand.of(customer, token, newPassword));
+            final Customer updatedCustomer =
+                    execute(CustomerPasswordResetCommand.of(customerByToken, tokenValue, newPassword));
 
-            final CustomerSignInResult signInResult = execute(CustomerSignInCommand.of(updatedCustomer.getEmail(), newPassword, null));
+            final CustomerSignInResult signInResult = execute(CustomerSignInCommand.of(email, newPassword));
             assertThat(signInResult.getCustomer().getId())
-                    .overridingErrorMessage("customer can sign in with the new password")
+                    .describedAs("customer can sign in with the new password")
```
Comment (lauraluiz) [inline sphere-models/src/it/java/io/sphere/sdk/customers/commands/CustomerPasswordResetCommandTest.java]: Hoping to see the differences between `describedAs`, `overridingErrorMessage` and `as` in your presentation (I could also check it now, but what's the fun of it? ;))

File: sphere-models/src/it/java/io/sphere/sdk/customers/commands/CustomerSignInCommandTest.java [hunk 2]
```diff
@@ -21,6 +38,53 @@ public void execution() throws Exception {
...
+                    anonymousCart.getLineItems().get(0).getQuantity());
+            assertThat(maxOfBothQuantities).isEqualTo(7);
+
+            final CustomerSignInResult result = execute(CustomerSignInCommand
+                    .of(customer.getEmail(), PASSWORD, anonymousCart.getId()));
+
+            final Cart mergeResultCart = result.getCart();
+            final LineItem mergeResultLineItem = mergeResultCart.getLineItems().get(0);
+            assertThat(mergeResultLineItem.getQuantity()).isEqualTo(maxOfBothQuantities).isEqualTo(7);
+            assertThat(mergeResultLineItem.getProductId())
```
Comment (lauraluiz) [inline sphere-models/src/it/java/io/sphere/sdk/customers/commands/CustomerSignInCommandTest.java]: Really? I have always expected it to be the sum of both: 10. Although I don't have any strong reason against/pro any form. Do you know why the maximum quantity?

Comment (schleichardt) [inline sphere-models/src/it/java/io/sphere/sdk/customers/commands/CustomerSignInCommandTest.java]: I think @gogregor knows.

File: sphere-models/src/it/java/io/sphere/sdk/customers/commands/CustomerPasswordResetCommandTest.java (hunk 1 above)
Comment (schleichardt) [inline sphere-models/src/it/java/io/sphere/sdk/customers/commands/CustomerPasswordResetCommandTest.java]: describedAs = as and will be added to the error message. overridingErrorMessage overrides it completely.

**Pull request merged**